whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -156,7 +156,10 @@ public Object map(final String requestUrl,
     }
 
     public Object mapSearch(final String requestUrl, final List<RpslObject> objects, final int maxResultSize) {
-        final SearchResult searchResult = new SearchResult();
+        final SearchResult searchResult = objects.isEmpty() ?
```

Comment:
Will this change return all 4 arrays if *any* search result is populated? We have to be careful only to return an empty array if the user has searched for the related type.

Reply:
I misunderstood the requirement. Last push should do it correctly

Reply:
I updated the PR description to match our interpretation of the RFC.